### PR TITLE
chore(http): REST API tokens support

### PR DESCRIPTION
**Pull request by puzpuzpuz**

### CODE REVIEW

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 28 / 31 (90.32%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/SqlUtil.java|18|21|85.71%|
|:large_blue_circle:|io/questdb/griffin/engine/table/LatestByValueListRecordCursor.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/auth/EllipticCurveAuthenticator.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/security/ReadOnlySecurityContext.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/std/datetime/microtime/Timestamps.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/security/AllowAllSecurityContext.java|2|2|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 28 / 31 (90.32%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/SqlUtil.java|18|21|85.71%|
|:large_blue_circle:|io/questdb/griffin/engine/table/LatestByValueListRecordCursor.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/auth/EllipticCurveAuthenticator.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/security/ReadOnlySecurityContext.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/std/datetime/microtime/Timestamps.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/security/AllowAllSecurityContext.java|2|2|100.00%|

File: core/src/main/java/io/questdb/griffin/engine/table/LatestByValueListRecordCursor.java [hunk 1]
```diff
@@ -135,15 +135,19 @@ public void of(DataFrameCursor dataFrameCursor, SqlExecutionContext executionCon
                 distinctSymbols++;
             }
             distinctSymbols = Math.max(0, distinctSymbols - excludedSymbolKeys.size());
-            rowIds.setCapacity(distinctSymbols);
+            if (distinctSymbols > 0) {
```
Comment (jerrinot) [inline core/src/main/java/io/questdb/griffin/engine/table/LatestByValueListRecordCursor.java:138]: I assume this is unrelated?

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 29 / 32 (90.62%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/SqlUtil.java|18|21|85.71%|
|:large_blue_circle:|io/questdb/griffin/engine/table/LatestByValueListRecordCursor.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/auth/EllipticCurveAuthenticator.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/security/ReadOnlySecurityContext.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/std/datetime/microtime/Timestamps.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlCodeGenerator.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/security/AllowAllSecurityContext.java|2|2|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 29 / 32 (90.62%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/SqlUtil.java|18|21|85.71%|
|:large_blue_circle:|io/questdb/griffin/engine/table/LatestByValueListRecordCursor.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/auth/EllipticCurveAuthenticator.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/security/ReadOnlySecurityContext.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/std/datetime/microtime/Timestamps.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlCodeGenerator.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/security/AllowAllSecurityContext.java|2|2|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 34 / 46 (73.91%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/ServerMain.java|1|10|10.00%|
|:large_blue_circle:|io/questdb/griffin/SqlUtil.java|18|21|85.71%|
|:large_blue_circle:|io/questdb/griffin/engine/table/LatestByValueListRecordCursor.java|5|5|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/auth/EllipticCurveAuthenticator.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpConnectionContext.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/security/DenyAllSecurityContext.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpConnectionContext.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/std/datetime/microtime/Timestamps.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlCodeGenerator.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/pgwire/PGConnectionContext.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/security/AllowAllSecurityContext.java|1|1|100.00%|

File: core/src/main/java/io/questdb/griffin/engine/table/LatestByValueListRecordCursor.java (hunk 1 above)
Comment (puzpuzpuz) [reply core/src/main/java/io/questdb/griffin/engine/table/LatestByValueListRecordCursor.java:138]: I've added a couple of tests to cover this change: bc8c1d39abfab5840e2b893f3a7459ad3311cb37

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 34 / 46 (73.91%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/ServerMain.java|1|10|10.00%|
|:large_blue_circle:|io/questdb/griffin/SqlUtil.java|18|21|85.71%|
|:large_blue_circle:|io/questdb/griffin/engine/table/LatestByValueListRecordCursor.java|5|5|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/auth/EllipticCurveAuthenticator.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpConnectionContext.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/security/DenyAllSecurityContext.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpConnectionContext.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/std/datetime/microtime/Timestamps.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlCodeGenerator.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/pgwire/PGConnectionContext.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/security/AllowAllSecurityContext.java|1|1|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 34 / 46 (73.91%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/ServerMain.java|1|10|10.00%|
|:large_blue_circle:|io/questdb/griffin/SqlUtil.java|18|21|85.71%|
|:large_blue_circle:|io/questdb/griffin/engine/table/LatestByValueListRecordCursor.java|5|5|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/auth/EllipticCurveAuthenticator.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpConnectionContext.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/security/DenyAllSecurityContext.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpConnectionContext.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/std/datetime/microtime/Timestamps.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlCodeGenerator.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/pgwire/PGConnectionContext.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/security/AllowAllSecurityContext.java|1|1|100.00%|

**Review by jerrinot — APPROVED** [review]

**Comment by puzpuzpuz** [comment]
@jerrinot thanks for the review!
